queue.js

```diff
@@ -0,0 +1,75 @@
+const {
+    SQSClient,
+    ReceiveMessageCommand,
+    DeleteMessageCommand,
+} = require('@aws-sdk/client-sqs');
+const retry = require('retry');
+
+const sqsClient = new SQSClient({
+    region: process.env.AWS_REGION,
+    credentials: {
+        accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    },
+});
+
+const queueURL = process.env.SQS_QUEUE_URL;
+
+const getMessage = async () => {
+    return new Promise((resolve, reject) => {
+        const receiveMessageCommand = new ReceiveMessageCommand({
+            QueueUrl: queueURL,
+            MaxNumberOfMessages: 1,
+            WaitTimeSeconds: 2,
+        });
+
+        const operation = retry.operation();
```

Comment:
how many time will you retry?

Reply:
getMessage will try to get a response from the queue infinitely currently. The retry logic here is mainly to address an issue where the queue becomes unavailable for some duration.

Reply:
In this case I actually think I should change it to not be infinite however to avoid the time between polling from getting too excessive and also so it would allow whatever error is causing the retries to propagate out. I'll redo it, thanks for pointing it out in your comment below.